amethyst_assets/src/asset.rs

```diff
@@ -85,6 +96,14 @@ where
         self.map.read().get(spec).map(Clone::clone)
     }
 
+    /// Accesses a cached asset, locking the internal `RwLock` to get read access to the hash map.
+    /// In case the asset exists, `f` gets called with a reference to the cached asset.
+    pub fn access<F: FnOnce(&T)>(&self, spec: &AssetSpec, f: F) {
```

Comment:
We probably ought to signal to the user of this function if a cached asset wasn't found.  Can we return a bool from here?  True if operation succeeded, false otherwise.  Sorry I didn't mention that on my first review.